### PR TITLE
Sameer/claim passport contract connect

### PR DESCRIPTION
**Pull request by crazzywizard**
### Purpose
Add ability to exchange ticket for a passport
* requires user to approve burn and then redeem for passport

### CODE REVIEW

File: lib/exchange.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    const tx = await contract.approve(
+      process.env.NEXT_PUBLIC_CRE8ORS_EXCHANGE_ADDRESS,
+      claimTicketId,
+    )
+    const res = await tx.wait()
+    return res
+  } catch (e) {
+    log.error(e)
+    return e
+  }
```
Comment (sweetmantech) [inline lib/exchange.ts]: bug: you are `catching` signature rejections here, so, in UX, user still progresses to `/success` screen.

suggestion: either
1. do not `try / catch` here, handle errors in caller
2. return an `isSuccess` variable

File: package.json [hunk 2]
```diff
@@ -47,7 +47,7 @@
     "twitter-api-v2": "^1.14.2",
     "typewriter-effect": "^2.20.1",
     "usehooks-ts": "^2.9.1",
-    "wagmi": "^0.8.10"
+    "wagmi": "^0.12.12"
```
Comment (sweetmantech) [inline package.json:50]: whoa BIG upgrade 🏗️ 

are we sure there aren't any unintended side-effects?

**Review by sweetmantech — COMMENTED** [review]
getting close, need to handle `user rejects` transactions

File: package.json (hunk 2 above)
Comment (crazzywizard) [reply package.json:50]: this is the version that is in reserve-ui repo, the biggest change was the removal of `allChains` function. I replaced it with own array of acceptable chains, mainly `mainnet, goerli, polygon, mumbai`

File: lib/exchange.ts (hunk 1 above)
Comment (crazzywizard) [reply lib/exchange.ts]: done! removed the try catch here

**Review by sweetmantech — APPROVED** [review]
ship it 🚢

**Pull request merged**